lib/grp.gi

```diff
@@ -528,13 +532,22 @@ InstallMethod( IsSolvableGroup,
     "generic method for groups",
     [ IsGroup ],
     function ( G )
-    local   S;          # derived series of <G>
+    local   S,          # derived series of <G>
+            isAbelian,  # true if <G> is abelian
+            isSolvable; # true if <G> is solvable
 
     # compute the derived series of <G>
     S := DerivedSeriesOfGroup( G );
 
     # the group is solvable if the derived series reaches the trivial group
-    return IsTrivial( S[ Length( S ) ] );
+    isSolvable := IsTrivial( S[ Length( S ) ] );
+
+    # set IsAbelian filter
+    isAbelian := isSolvable and Length( S ) <= 2;
+    Assert(2, IsAbelian(G) = isAbelian);
```

Comment:
I did not know about the assertion level being increased for tests, thank you. Increased it to 3. There are several other places in grp.gi where I think very similar lines exist with assertion level 2.